toolchain/sem_ir/node.h

```diff
@@ -14,542 +14,118 @@
-  using HasParseNode = std::false_type;
-
-  CrossReferenceIRId ir_id;
-  NodeId node_id;
-};
-
-struct Dereference {
-  NodeId pointer_id;
-};
-
-struct FunctionDeclaration {
-  FunctionId function_id;
-};
-
-// Finalizes the initialization of `dest_id` from the initializer expression
-// `src_id`, by performing a final copy from source to destination, for types
-// whose initialization is not in-place.
-struct InitializeFrom {
-  NodeId src_id;
-  NodeId dest_id;
-};
-
-struct IntegerLiteral {
-  IntegerId integer_id;
-};
-
-struct NameReference {
-  StringId name_id;
-  NodeId value_id;
-};
-
-struct Namespace {
-  NameScopeId name_scope_id;
-};
-
-struct NoOp {
-  using HasType = std::false_type;
-};
-
-struct Parameter {
-  StringId name_id;
-};
-
-struct PointerType {
-  TypeId pointee_id;
-};
-
-struct RealLiteral {
-  RealId real_id;
-};
-
-struct Return {
-  using HasType = std::false_type;
-};
-
-struct ReturnExpression {
-  using HasType = std::false_type;
-
-  NodeId expr_id;
-};
-
-struct SpliceBlock {
-  NodeBlockId block_id;
-  NodeId result_id;
-};
-
-struct StringLiteral {
-  StringId string_id;
-};
-
-struct StructAccess {
-  NodeId struct_id;
-  MemberIndex index;
-};
-
-struct StructInit {
-  NodeId src_id;
-  NodeBlockId elements_id;
-};
-
-struct StructLiteral {
-  NodeBlockId elements_id;
-};
-
-struct StructType {
-  NodeBlockId fields_id;
-};
-
-struct StructTypeField {
-  using HasType = std::false_type;
-
-  StringId name_id;
-  TypeId type_id;
-};
-
-struct StructValue {
-  NodeId src_id;
-  NodeBlockId elements_id;
-};
-
-struct Temporary {
-  NodeId storage_id;
-  NodeId init_id;
-};
-
-struct TemporaryStorage {};
-
-struct TupleAccess {
-  NodeId tuple_id;
-  MemberIndex index;
-};
-
-struct TupleIndex {
-  NodeId tuple_id;
-  NodeId index_id;
-};
-
-struct TupleInit {
-  NodeId src_id;
-  NodeBlockId elements_id;
-};
-
-struct TupleLiteral {
-  NodeBlockId elements_id;
-};
-
-struct TupleType {
-  TypeBlockId elements_id;
-};
-
-struct TupleValue {
-  NodeId src_id;
-  NodeBlockId elements_id;
-};
-
-struct UnaryOperatorNot {
-  NodeId operand_id;
-};
-
-struct ValueAsReference {
-  NodeId value_id;
-};
-
-struct VarStorage {
-  StringId name_id;
-};
-}  // namespace NodeData
-
-template <NodeKind::RawEnumType KindT, typename DataT>
-struct TypedNode;
-
-// The standard structure for Node. This is trying to provide a minimal
-// amount of information for a node:
+// A type-erased representation of a SemIR node, that may be constructed from
+// the specific kinds of node defined in `typed_nodes.h`. This provides access
+// to common fields present on most or all kinds of nodes:
 //
 // - `parse_node` for error placement.
 // - `kind` for run-time logic when the input Kind is unknown.
 // - `type_id` for quick type checking.
-// - Up to two Kind-specific members.
 //
-// To create a specific kind of `Node`, use the appropriate `TypedNode`
-// constructor. A `TypedNode` implicitly converts to a `Node`.
```

Comment:
Perhaps revive this comment with a pointer to `typed_nodes.h`?

Reply:
The class comment already starts with: "that may be constructed from the specific kinds of node defined in `typed_nodes.h`", do you want more than that?